File: GXpert/GXpert.Web/Modules/ServerTypes/Institute/InstituteDivisionForm.ts
import { LookupEditor, ServiceLookupEditor, StringEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface InstituteDivisionForm {
    InstituteId: LookupEditor;
    DepartmentId: LookupEditor;
    CourseId: ServiceLookupEditor;
    ClassId: LookupEditor;
    SemesterId: ServiceLookupEditor;
    Title: StringEditor;
    Division: StringEditor;
    ClassTeacherId: LookupEditor;
    AcademicYearId: LookupEditor;
}

export class InstituteDivisionForm extends PrefixedContext {
    static readonly formKey = 'Institute.InstituteDivision';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!InstituteDivisionForm.init)  {
            InstituteDivisionForm.init = true;

            var w0 = LookupEditor;
            var w1 = ServiceLookupEditor;
            var w2 = StringEditor;

            initFormType(InstituteDivisionForm, [
                'InstituteId', w0,
                'DepartmentId', w0,
                'CourseId', w1,
                'ClassId', w0,
                'SemesterId', w1,
                'Title', w2,
                'Division', w2,
                'ClassTeacherId', w0,
                'AcademicYearId', w0
            ]);
        }
    }
}

File: GXpert/GXpert.Web/Modules/Institute/InstituteDivision/InstituteDivisionForm.cs
using Serenity.ComponentModel;
using System;

namespace GXpert.Institute.Forms;

[FormScript("Institute.InstituteDivision")]
[BasedOnRow(typeof(InstituteDivisionRow), CheckNames = true)]
public class InstituteDivisionForm
{
    public int InstituteId { get; set; }
    [HalfWidth]
    public int DepartmentId { get; set; }
    [HalfWidth]
    public int CourseId { get; set; }

    [HalfWidth]
    public int ClassId { get; set; }

    [HalfWidth]
    public int SemesterId { get; set; }
    [HalfWidth]
    public string Title { get; set; }
    [HalfWidth]
    public string Division { get; set; }
    [HalfWidth]

    public int ClassTeacherId { get; set; }

    [HalfWidth]
    public int AcademicYearId { get; set; }
    //public int MediumId { get; set; }
    
}